kyo-prelude/shared/src/main/scala/kyo/Var.scala

```diff
@@ -60,6 +60,21 @@ object Var:
     inline def set[V](inline value: V)(using inline tag: Tag[Var[V]], inline frame: Frame): V < Var[V] =
         ArrowEffect.suspend[Unit](tag, value: Op[V])
 
+    /** Sets a new value and then executes another computation.
+      *
+      * @param value
+      *   The new value to set in the Var
+      * @param f
+      *   The computation to execute after setting the value
+      * @return
+      *   The result of the computation after setting the new value
+      */
+    inline def setAndThen[V, A, S](inline value: V)(inline f: => A < S)(using
```

Comment:
that's a good point. I added it actually as a minor performance optimization since it reduces from two allocations in the equivalent `Var.set(..).andThen` to a single one. I'd be ok with making it `private[kyo]` but we might eventually need to provide this kind of API to library developers for best performance.